### ISSUE TITLE
"Couldn't find template for digesting" error with js.erb views

### ISSUE THREAD

**Issue by saneef** [rails/rails#29423]
### Steps to reproduce

I have created the code and steps to reproduce the problem at https://github.com/saneef/reproduce-couldnt-find-template-for-digesting

### Expected behavior

Should do fragment caching without any issues 

### Actual behavior

Error: `Couldn't find template for digesting`

### System configuration
**Rails version**: 5.1.1

**Ruby version**: 2.3.3

May be similar to #15255, but suggested hack like using full paths didn't fix the issue.

**Comment by jfyzhao** [comment]
+1

**Comment by Agiley** [comment]
Somehow you need to specify the full path to the partial including ".html.erb" (or whatever template engine you're using) if you try to render from a js.erb-file. If I do that I no longer get "Couldn't find template for digesting: hotels/_hotel".

Examples:
render(partial: @hotels, cached: true) = 
"Couldn't find template for digesting: hotels/_hotel"

render(partial: 'hotels/hotel', collection: @hotels, cached: true) = 
"Couldn't find template for digesting: hotels/_hotel"

render(partial: 'hotels/hotel.html.erb', collection: @hotels, cached: true) =
No errors

**Comment by savroff** [comment]
@saneef  because you are using `js.erb` you just need to have `_icon.js.erb`, or if you using same partial for `:html` and `:js`, you can just rename your partial to `_icon.erb`.

This will fix the problem with your caching fragment. 

I don't think it will be a good idea to do a fallback on `:html`, but I think we need to put this information somewhere in caching documentation.

**Comment by savroff** [comment]
I will check more if it is an easy way to do fallback and come back soon

**Comment by saneef** [comment]
@Agiley Cool! Using the full path with extensions works.

@savroff I'll stick with only `.erb` extension approach. It would be great to document this somewhere and let us know if you this approach have any repercussions.

**Comment by savroff** [comment]
I'm not good with words :) Maybe someone can put it logically to documentation in a way like this?

#### Shared partial caching

In case if you need to cache the same partial in `html.erb` and `js.erb` files,
you can remove extension from a partial file name  `_product.erb`.

**Comment by gwincr11** [comment]
How does this sound?

It is possible to share partials and associated caching between files with different mime types. For example shared partial caching allows template writers to share a partial between html and javascript files. Because of the way templates are collected in the template resolver file paths that only include the template language extension, not the mime type, can be used across mime types. So a request for an html file or js file that includes the following code:
```ruby
render(partial: 'hotels/hotel', collection: @hotels, cached: true)
```
Will load a file named `hotels/hotel.erb`.

**Comment by HarlemSquirrel** [comment]
I ran into this issue on Rails 5.1.3 and fixed it with the following. I think rails is looking for another partial of the same type, `.js` before falling back to the only partial I have with that name. 

```ruby
render partial: 'partial_name.html', cached: true
```

**Comment by gwincr11** [comment]
I am wondering if this can be closed as #29964 fixes the docs.

**Comment by saneef** [comment]
Thanks @gwincr11. Closing since the docs updated with usage on #29964.
